src/Array.mo

```diff
@@ -697,5 +697,25 @@ module {
   /// Runtime: O(1)
   ///
   /// Space: O(1)
-  public func keys<X>(array : [X]) : I.Iter<Nat> = array.keys()
-}
+  public func keys<X>(array : [X]) : I.Iter<Nat> = array.keys();
+
+
+  /// Returns a new subarray from the given array from the start index to the end index (inclusive)
+  ///
+  /// Limitations: Traps if the start or end indices are out of bounds
+  ///
+  /// ```motoko include=import
+  ///
+  /// let array = [1,2,3,4,5];
+  /// let subArray = Array.subArray<Nat>(array, 2, 4);
+  ///
+  /// subArray; // [3,4,5];
+  ///
+  /// Runtime: O(endIndex - startIndex);
+  /// Space: O(endIndex - startIndex);
+  public func subArray<T>(arr: [T], startIndex: Nat, endIndex: Nat): [T] {
+    tabulate<T>(endIndex - startIndex + 1, func(i) { 
+      arr[startIndex + i] 
```

Comment:
```suggestion
    tabulate<X>(count, func(i) { 
      arr[start + i] 
```
Also, I would do some bounds checcking (similar to `subBuffer` in `Buffer`) so that it traps on illegal inputs